primo/opt_model/result_parser.py

```diff
@@ -0,0 +1,776 @@
+
+    @property
+    def average_age(self):
+        """
+        Returns average age of the wells in the project
+        """
+        return self.well_data[self._col_names.age].mean()
+
+    @property
+    def age_range(self):
+        """
+        Returns the range of the age of the project
+        """
+        return (
+            self.well_data[self._col_names.age].max()
+            - self.well_data[self._col_names.age].min()
+        )
+
+    @property
+    def average_depth(self):
+        """
+        Returns the average depth of the project
+        """
+        return self.well_data[self._col_names.depth].mean()
+
+    @property
+    def depth_range(self):
+        """
+        Returns the range of the depth of the project
+        """
+        return (
+            self.well_data[self._col_names.depth].max()
+            - self.well_data[self._col_names.depth].min()
+        )
+
+    @property
+    def avg_elevation_delta(self):
+        """
+        Returns the average elevation delta of the project
+        """
+        col_name = self._col_names.elevation_delta
+        self._check_column_exists(col_name)
+        return calculate_average(
+            self.well_data.data,
+            self._col_names.elevation_delta,
+            estimation_method="yes",
+        )
+
+    @property
+    def centroid(self):
+        """
+        Returns the centroid of the project
+        """
+        return tuple(
+            np.round(
+                np.mean(
+                    self.well_data.data[
+                        [self._col_names.latitude, self._col_names.longitude]
+                    ].values,
+                    axis=0,
+                ),
+                6,
+            )
+        )
+
+    @property
+    def avg_dist_to_road(self):
+        """
+        Returns the average distance to road for a project
+        """
+        col_name = self._col_names.dist_to_road
+        self._check_column_exists(col_name)
+        return calculate_average(self.well_data.data, self._col_names.dist_to_road)
+
+    @property
+    def num_unique_owners(self):
+        """
+        Returns the number of well owners for a project
+        """
+        col_name = self._col_names.operator_name
+        self._check_column_exists(col_name)
+        return len(set(self.well_data[col_name].values))
+
+    @property
+    def impact_score(self):
+        """
+        Returns the average priority score for the project
+        """
+        if not hasattr(self._col_names, "priority_score"):
+            raise AttributeError(
+                "The priority score has not been computed for the Well Data"
+            )
+        return self.well_data[self._col_names.priority_score].mean()
+
+    @property
+    def accessibility_score(self):
+        """
+        Returns the accessibility score and the total weight of the accessibility score for a project
+        """
+        names_attributes = [i for i in dir(self) if "eff_score" in i]

```

Comment:
Similarly not a fan of i although I understand it is a basic iterator

Reply:
Changed. It is more clear.